services/frontend/src/components/SearchBar.js

```diff
@@ -1,18 +1,61 @@
-const SearchBar = () => {
+import { useState } from "react";
+import { PropTypes } from "prop-types";
+
+const SearchBar = ({ handleUserSearch }) => {
+  const [userInput, setUserInput] = useState("");
+
+  const handleUserInputChange = (event) => {
+    setUserInput(event.target.value);
+  };
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
```

Comment:
By default an HTML form submission will trigger the browser to reload. In this case we don't want that, React will handle updating the content without reloading. `event.preventDefault()` overrides this behavior and lets React do its thing.

The sites you make with modern frameworks are often called "single page apps" because you only every load the single root page. After that everything is loaded dynamically by React. You can mimic multi-page behavior (like allowing the user to load `mypage.com/about`) by using a tool like [React Router](https://reactrouter.com/web/guides/quick-start).